docs/basics/networkbehaviour.md

```diff
@@ -108,3 +108,130 @@ If you override the virtual 'OnDestroy' method it is important to alway invoke t
 
 `NetworkBehaviour` handles other destroy clean up tasks and requires that you invoke the base `OnDestroy` method to operate properly.
 :::
+
+### NetworkBehaviour Pre-Spawn Synchronization
+
+There could be scenarios where you need to include additional configuration data or use a `NetworkBehaviour` to configure some non-netcode related component (or the like) prior to a `NetworkObject` being spawned. This can be particularly critical if you want specific settings applied prior to `NetworkBehaviour.OnNetworkSpawn` being invoked. When a client is synchronizing with an existing network session, this can become problematic as messaging requires a client to be fully synchronized before you really know "it is safe" to send the message and even if you send a message there is the latency involved in the whole process that might not be convenient and could require additional specialized code to account for this.
+
+`NetworkBehaviour.OnSynchronize` allows you to write and read custom serialized data during the `NetworkObject` serialization process.  
+
+There are two cases where `NetworkObject` synchronization occurs:
+- When dynamically spawning a `NetworkObject`.
+- When a client is being synchronized after connection approval 
+  - i.e. Full synchronization of the `NetworkObject`s and scenes.
+
+:::info
+If you haven't already become familiar with the [`INetworkSerializable` interface](../advanced-topics/serialization/inetworkserializable.md), then you might read up on that before proceeding as `NetworkBehaviour.OnSynchronize` as it follows a very similar usage pattern.
+:::
+
```

Comment:
There could be scenarios where you need to include additional configuration data or use a `NetworkBehaviour` to configure some non-Netcode related component (or the like) before a `NetworkObject` spawns. Doing so can be particularly critical if you want specific settings applied before invoking `NetworkBehaviour.OnNetworkSpawn`. A client synchronizing with an existing network session can become problematic because messaging requires a client to synchronize fully before you know it's "safe" to send the message. And, even if you send the message, it introduces (possibly) inconvenient latency, and you might need to add additional specialized code to account for the latency.

`NetworkBehaviour.OnSynchronize` allows you to write and read custom serialized data during the `NetworkObject` serialization process.  

There are two cases where `NetworkObject` synchronization occurs:
- When dynamically spawning a `NetworkObject`.
- When synchronizing a client after connection approval (that is, full synchronization of the `NetworkObject's and scenes.

:::info
If you're unfamiliar with the [`INetworkSerializable` interface](../advanced-topics/serialization/inetworkserializable.md), read up on it before proceeding because `NetworkBehaviour.OnSynchronize` as it follows a very similar usage pattern.
:::